src/lib.rs

```diff
@@ -36,6 +36,25 @@ pub enum CowArc<'a, T: ?Sized + 'static> {
     Owned(Arc<T>),
 }
 
+impl<'a: 'static, T: 'static> CowArc<'static, T> {
+    /// Creates a new [`CowArc::Owned`] from a value.
+    ///
+    /// This is simply a convenience method;
+    /// the value will be wrapped in an [`Arc`].
+    ///
+    /// Note that `T` must be [`Sized`]: use the enum constructor directly if `T` is unsized.
```

Comment:
Could we make `T: ?Sized` and then have a where clause on this function that requires `T: Sized`? That should allow the other constructor to work with unsized inner values